processor/batchprocessor/batch_processor_test.go

```diff
@@ -80,6 +80,91 @@ func TestProcessorLifecycle(t *testing.T) {
 	}
 }
 
+func BenchmarkBatchTraceProcessor(b *testing.B) {
+	b.StopTimer()
+	cfg := Config{
+		Timeout:                  100 * time.Millisecond,
+		SendBatchSize:            1000,
+		MetadataCardinalityLimit: defaultMetadataCardinalityLimit,
+	}
+	runTraceProcessorWoMetadataBenchmark(b, cfg)
+}
+
+func runTraceProcessorWoMetadataBenchmark(b *testing.B, cfg Config) {
+	ctx := context.Background()
+
+	sink := new(consumertest.TracesSink)
+	cfg.SendBatchSize = 128
+	creationSet := processortest.NewNopSettings()
+	creationSet.MetricsLevel = configtelemetry.LevelDetailed
+	batcher, err := newBatchTracesProcessor(creationSet, sink, &cfg)
+	require.NoError(b, err)
+	require.NoError(b, batcher.Start(context.Background(), componenttest.NewNopHost()))
+
+	b.StartTimer()
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			requestCount := 100
+			spansPerRequest := 10
+			sentResourceSpans := ptrace.NewTraces().ResourceSpans()
+			for requestNum := 0; requestNum < requestCount; requestNum++ {
+				td := testdata.GenerateTraces(spansPerRequest)
```

Comment:
Please move `td` building outside. We don't need to benchmark it. We need to benchmark `batcher.ConsumeTraces(context.Background(), td))` only